plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/debug/ui/launchConfigurations/LaunchConfigurationsDialog.java

```diff
@@ -206,6 +206,12 @@ public String errorMessageWhile() {
 		public String errorMessageUntil() {
 			return "Tree does not has focus or TreeItem '"+item.getText()+"' is not selected";
 		}
+
+		@Override
```

Comment:
So, we should keep interface WaitCondition intact (not forcing classes, that use interface, to implement getResult), and then AbstractWaitCondition class having default implementation of getResult and possibly overriding the method if child class needs to do so? Aka, remove getResult from WaitCondition and have it with default return in AbstractWaitCondition?